FWCore/ParameterSet/interface/ParameterDescriptionCases.h

```diff
@@ -39,64 +39,64 @@ namespace edm {
 
-    void insert(T caseValue, std::auto_ptr<ParameterDescriptionNode> node) {
+    void insert(T caseValue, std::unique_ptr<ParameterDescriptionNode> node) {
       std::pair<T, edm::value_ptr<ParameterDescriptionNode> > casePair(caseValue,edm::value_ptr<ParameterDescriptionNode>());
       std::pair<typename CaseMap::iterator,bool> status;
       status = caseMap_->insert(casePair);
-      (*caseMap_)[caseValue] = node;
+      (*caseMap_)[caseValue] = std::move(node);
       if (status.second == false) duplicateCaseValues_ = true;
     }
 
-    std::auto_ptr<CaseMap> caseMap() { return caseMap_; }
+    std::unique_ptr<CaseMap> caseMap() { return std::move(caseMap_); }
     bool duplicateCaseValues() const { return duplicateCaseValues_; }
 
   private:
 
     friend
-    std::auto_ptr<ParameterDescriptionCases<bool> >
+    std::unique_ptr<ParameterDescriptionCases<bool> >
     operator>>(bool caseValue,
-               std::auto_ptr<ParameterDescriptionNode> node);
+               std::unique_ptr<ParameterDescriptionNode> node);
 
     friend
-    std::auto_ptr<ParameterDescriptionCases<int> >
+    std::unique_ptr<ParameterDescriptionCases<int> >
     operator>>(int caseValue,
-               std::auto_ptr<ParameterDescriptionNode> node);
+               std::unique_ptr<ParameterDescriptionNode> node);
 
     friend
-    std::auto_ptr<ParameterDescriptionCases<std::string> >
+    std::unique_ptr<ParameterDescriptionCases<std::string> >
     operator>>(std::string const& caseValue,
-               std::auto_ptr<ParameterDescriptionNode> node);
+               std::unique_ptr<ParameterDescriptionNode> node);
 
     friend
-    std::auto_ptr<ParameterDescriptionCases<std::string> >
+    std::unique_ptr<ParameterDescriptionCases<std::string> >
     operator>>(char const* caseValue,
-               std::auto_ptr<ParameterDescriptionNode> node);
+               std::unique_ptr<ParameterDescriptionNode> node);
 
     // The constructor is intentionally private so that only the operator>> functions
     // can create these. 
-    ParameterDescriptionCases(T const& caseValue, std::auto_ptr<ParameterDescriptionNode> node) :
+    ParameterDescriptionCases(T const& caseValue, std::unique_ptr<ParameterDescriptionNode> node) :
       caseMap_(new CaseMap),
       duplicateCaseValues_(false)
     {
       std::pair<T, edm::value_ptr<ParameterDescriptionNode> > casePair(caseValue,edm::value_ptr<ParameterDescriptionNode>());
       caseMap_->insert(casePair);
-      (*caseMap_)[caseValue] = node;
+      (*caseMap_)[caseValue] = std::move(node);
     }
 
-    std::auto_ptr<CaseMap> caseMap_;
+    std::unique_ptr<CaseMap> caseMap_;
     bool duplicateCaseValues_;
   };
 
-  std::auto_ptr<ParameterDescriptionCases<bool> >
-  operator||(std::auto_ptr<ParameterDescriptionCases<bool> >,
-             std::auto_ptr<ParameterDescriptionCases<bool> >);
+  std::unique_ptr<ParameterDescriptionCases<bool> >
```

Comment:
I think it is OK. It is already a sink for the object. Also this interface is not intended to be used outside the Framework, so I would think the compiler and tests would note any problems.